src/charm.py

```diff
@@ -425,6 +425,19 @@ def wait_for_rollout(self, name, namespace="kube-system", timeout=300):
             "rollout", "status", "-n", namespace, name, "--timeout", str(timeout) + "s"
         )
 
+    def get_peer_data_with_key(self, data_key):
```

Comment:
This method is fine with maybe a name change.  But I wouldn't leave the `get_service_cidr(...)` method with a single line statement in it.  Just find `self.get_service_cidr()` and replace with `self.kube_ovn_peer_data("service-cidr")`.  

Be glad you didn't let me write this or i'd feature creep some object in there that holds all the logic for handling the relation.

```suggestion
    def kube_ovn_peer_data(self, key):
```